docs/docs/best-practices.md

```diff
@@ -55,3 +55,32 @@ enum Modes {
 ```
 
 When using an `Enum`, prefer using a string Enum.
+
+### Producers:
+- only one producer per file
+- they should have a descriptive name (camelCase) and the file should have the same name
+- they should do only one thing
+- they should use `get` instead of `observe` if you don’t need that value to retrigger the producer
+- in the folder structure they should stay close to the components that use them. 
+- if there are more than one producer for a component the propper way of **importing** them should be to by creating an `index.js/ts` file in the producers folder and export them all there and in the component write like this
+``` import * as producers from "./producers";
+...
+<Component>.producers(Object.values(producers)); 
+```
+- if you use typescript keep using the types throughout the entire producer (use type casting if needed)
+
+### Components:
+- only one component per file
+- they should have a descriptive name (PascalCase) and the file should have the same name
+- they should do only one thing
+- they should use `get` instead of `observe` if you don’t need that value to retrigger the component
```

Comment:
I think `get` should be avoided in views. Views should be.... views of data - their responsibility is to always be up to date.

If you can think of a use case where you would need `get` in a component, please let me know, that would a be an interesting conversation!

Reply:
Maybe if you have some semi-permanent values in the initial state for a profile/theme for exemple

Reply:
Using `get` should be reserved to events that happen within the view e.g. `onClick`.
```
const A: view = ({
  foo= observe.foo,
  updateBar = update.bar,
  getId = get.selectedId
}) => {
  return <div onClick={() => updateBar.set(getId.value()) }>CLick me</div>
}
```